src/pages/Menu.js

```diff
@@ -11,194 +11,203 @@ import burgerBaconBuddy from '../assets/images/burger-bacon-buddy.png';
 import burgerSpicy from '../assets/images/burger-spicy.png';
 import burgerClassic from '../assets/images/burger-classic.png';
 
-class MenuClass extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      loading: true,
-      meals: [],
-      visibleMeals: [],
-      quantities: {},
-      currentPage: 1,
-      itemsPerPage: 6,
-      hasMore: true
-    };
-    
-    this.handleQuantityChange = this.handleQuantityChange.bind(this);
-    this.handleAddToCart = this.handleAddToCart.bind(this);
-    this.loadMoreItems = this.loadMoreItems.bind(this);
-    this.handleCategoryClick = this.handleCategoryClick.bind(this);
-  }
+const Menu = () => {
+  const { addToCart } = useCart();
+  const [loading, setLoading] = useState(true);
+  const [meals, setMeals] = useState([]);
+  const [visibleMeals, setVisibleMeals] = useState([]);
+  const [quantities, setQuantities] = useState({});
+  const [currentPage, setCurrentPage] = useState(1);
+  const [itemsPerPage] = useState(6);
+  const [hasMore, setHasMore] = useState(true);
+  const [category, setCategory] = useState('Dessert');
   
-  componentDidMount() {
+  useEffect(() => {
     document.title = "Menu";
-    this.fetchMeals();
-  }
+    fetchMeals();
+  }, []);
+  
+  useEffect(() => {
+    if (meals.length > 0) {
+      const filteredMeals = meals.filter(meal => meal.category === category);
+      setVisibleMeals(filteredMeals.slice(0, currentPage * itemsPerPage));
+      setHasMore(filteredMeals.length > currentPage * itemsPerPage);
+    }
+  }, [category, meals, currentPage, itemsPerPage]);
   
-  fetchMeals() {
+  const fetchMeals = () => {
     fetch('https://65de35f3dccfcd562f5691bb.mockapi.io/api/v1/meals')
```

Comment:
Could add filter to request and in that case there will be not need in filtering on FE